src/app/[locale]/viewer/_components/transfer/TransferDialog.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+    Box,
+    Dialog,
+    DialogContent,
+    DialogProps,
+    Divider,
+    IconButton,
+    Typography
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import {
+    TargetRepositoryFormData,
+    TargetRespositories
+} from 'app/[locale]/viewer/_components/transfer/TargetRespositories';
+import { FormattedMessage } from 'react-intl';
+import { messages } from 'lib/i18n/localization';
+import { AssetAdministrationShell } from '@aas-core-works/aas-core3.0-typescript/types';
+import { useEffect, useState } from 'react';
+import { SubmodelOrIdReference, useAasState, useSubmodelState } from 'components/contexts/CurrentAasContext';
+import { transferAasWithSubmodels, TransferDto } from 'lib/services/transfer-service/transferActions';
+import { useNotificationSpawner } from 'lib/hooks/UseNotificationSpawner';
+
+export type TransferFormModel = {
+    targetAasRepositoryBaseUrl?: string;
+    targetSubmodelRepositoryBaseUrl?: string;
+    aas?: AssetAdministrationShell | null;
+    submodels?: SubmodelOrIdReference[];
+};
+
+export function TransferDialog(props: DialogProps) {
+    const [transferDto, setTransferDto] = useState<TransferFormModel>();
+    const [submodelsFromContext,] = useSubmodelState();
+    const [aasFromContext,] = useAasState();
+    const notificationSpawner = useNotificationSpawner();
+
+    useEffect(() => {
+        const transferDtoTemp = { ...transferDto, aas: aasFromContext, submodels: submodelsFromContext }
+
+        setTransferDto(transferDtoTemp)
+    }, [])
+
+    const handleSubmitRepositoryStep = async (values: TargetRepositoryFormData, redirectToNew: boolean) => {
+        if (!values.repository || !aasFromContext || !values.repository) {
+            return;
+        }
+
+        // This state can be used later to hold the data of multiple steps
+        const transferDtoTemp = {
```

Comment:
this feels a bit duplicated to the effect above.
Maybe merge them together, if possible?